src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/MandateForChange.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.hmppstier.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppstier.client.CommunityApiClient
+import uk.gov.justice.digital.hmpps.hmppstier.client.Conviction
+import uk.gov.justice.digital.hmpps.hmppstier.client.Sentence
+
+@Service
+class MandateForChange(
+  private val communityApiClient: CommunityApiClient
+) {
+  fun hasNoMandate(crn: String, convictions: Collection<Conviction>): Boolean =
+    convictions
+      .filter { it.sentence?.terminationDate == null }
+      .let { activeConvictions ->
+        activeConvictions.none {
+          it.sentence != null && (isCustodialSentence(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId))
+        }
+      }.also { log.debug("Has Mandate for change: $it") }
+
+  private fun isCustodialSentence(sentence: Sentence) =
+    sentence.sentenceType.code in custodialSentences
+
+  private fun hasNonRestrictiveRequirements(crn: String, convictionId: Long): Boolean =
+    communityApiClient.getRequirements(crn, convictionId)
+      .filter { req -> req.requirementTypeMainCategory.description != "Unpaid Work" }
```

Comment:
Is there a ref data code for this? I appreciate "Unpaid Work" isn't likely to change but matching on human strings makes me uncomfortable.

Reply:
`{"requirements":[{"requirementId":1502375618,"commencementDate":"2019-10-11","startDate":"2019-10-11","terminationDate":"2020-11-21","expectedStartDate":"2019-10-11","active":false,"requirementTypeSubCategory":{"code":"W01","description":"Regular"},"requirementTypeMainCategory":{"code":"W","description":"Unpaid Work"},"terminationReason":{"code":"51","description":"Requirement Completed"},"length":200,"lengthUnit":"Hours","restrictive":false}]}`

I can change it to req.requirementTypeMainCategory.Code != W - does that help?